src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -198,6 +191,11 @@ public function addContent($content, $type = null)
      */
     public function addHtmlContent($content, $charset = 'UTF-8')
     {
+        // Use HTML5 parser if the content is HTML5 and the library is available
+        if (!$this->html5Parser && class_exists(HTML5::class) && 0 === stripos(ltrim($content), '<!DOCTYPE html>')) {
```

Comment:
how about using substr instead? content can be pretty large so strpos miss could be slow